pkg/auth/providers/azure/azure_provider_test.go

```diff
@@ -225,3 +228,73 @@ func TestTransformToAuthProvider(t *testing.T) {
 		})
 	}
 }
+
+func TestProcessConfigOnTestAndApply(t *testing.T) {
+	t.Parallel()
+
+	tests := []struct {
+		name               string
+		current            *v3.AzureADConfig
+		new                *v3.AzureADConfig
+		annotationExpected bool
+	}{
+		{
```

Comment:
I think you are missing a test case for if the current config is enabled and has the new flow annotation (in this case we don't want to set the annotation). 

This is also leaving out cases where the new config has an annotation and we make sure that we overrode it and didn't just use what was already there. I would recommend at least one test case where new has an annotation for this value and we forcibly override it to say that it is enabled.

Reply:
That one I do have, but I noticed that am missing a case when Azure AD is set up for the first time, and the current one has the annotation, meaning it had been set on Rancher startup.

Reply:
Also, I don't think we should rely on whether the proposed one has the annotation, right? The backend now figures out on its own if the new config must have it or not.

Reply:
> That one I do have, but I noticed that am missing a case when Azure AD is set up for the first time, and the current one has the annotation, meaning it had been set on Rancher startup.

Near as I can tell there are 4 test cases here:
- Current is enabled, and has the annotation set to the right value (set the annotation on proposed)
- Current is enabled, but does not have the annotation set (don't set the annotation on proposed)
- Current is not enabled, but has the new flow annotation (set the annotation on proposed)
- Current is not enabled, and does not have the new flow annotation (set the annotation on proposed)

That's not counting any "part of input is nil" cases. As long as we have those 4, we should be good. I'll give it another look.

> Also, I don't think we should rely on whether the proposed one has the annotation, right? The backend now figures out on its own if the new config must have it or not.

My point here is that you are overwriting what is present on the proposed config. It's possible to change the code to leave what's on the proposed config if the user supplied this annotation (for example if the user changed this to use the old side). If this happened, all of your test cases would still pass, but we wouldn't want that to happen.